core/src/avm1/globals/system.rs

```diff
@@ -292,28 +341,415 @@ pub struct SystemProperties {
+                Some("US".to_string())
+            }
+        } else {
+            None
+        };
+
+        #[cfg(not(target_family = "wasm"))]
+        fn get_os() -> OperatingSystem {
+            pub const OS: &str = std::env::consts::OS;
+
+            return match OS {
+                "linux" | "android" | "freebsd" | "dragonfly" | "netbsd" | "openbsd"
+                | "solaris" => OperatingSystem::Linux,
+                "macos" | "ios" => OperatingSystem::MacOs,
+                "windows" => {
+                    #[allow(clippy::zero_prefixed_literal)]
+                    fn match_windows_version(
+                        major_version: u64,
+                        minor_version: u64,
+                        build_version: u64,
+                        server_version: bool,
+                    ) -> OperatingSystem {
+                        let version = (major_version, minor_version);
+                        match version {
+                            (10, 0) | (6, 4) | (6, 3) | (6, 2) => OperatingSystem::Windows8,
+                            (6, 1) => {
+                                if server_version {
+                                    OperatingSystem::WindowsServer2008R2
+                                } else {
+                                    OperatingSystem::Windows7
+                                }
+                            }
+                            (6, 0) => {
+                                if server_version {
+                                    OperatingSystem::WindowsServer2008
+                                } else {
+                                    OperatingSystem::WindowsVista
+                                }
+                            }
+                            (5, 2) => {
+                                if server_version {
+                                    OperatingSystem::WindowsServer2003R2
+                                } else {
+                                    OperatingSystem::WindowsXp
+                                }
+                            }
+                            (5, 1) => OperatingSystem::WindowsXp,
+                            (5, 0) => OperatingSystem::Windows2k,
+                            (4, 1) | (4, 10) => OperatingSystem::Windows98,
+                            (4, 0) | (4, 03) => {
+                                // 950 - 1216 are Windows 95, 1381 is Windows NT 4.0
+                                if build_version <= 1300 {
+                                    OperatingSystem::Windows95
+                                } else {
+                                    OperatingSystem::WindowsNt
+                                }
+                            }
+                            (3, 51) | (3, 50) | (3, 5) | (3, 10) | (3, 1) => {
+                                OperatingSystem::WindowsNt
+                            }
+                            _ => OperatingSystem::WindowsUnknown,
+                        }
```

Comment:
Might be a bit overkill as I don't even think Rust can compile for these 😆

Reply:
Apparently compiled Rust doesn’t work on Windows XP or earlier without much manual work, so I’m assuming there won’t be a Ruffle version for those. And I can see that it makes sense to remove this for versions where Ruffle doesn’t work on (or won’t be changed to work on at some point).
I’ve included the fallback to use ver on the shells also just to increase compatibility to older Windows systems (before Windows 2000), which aren’t supported by os_version, so this is also not necessary, since Ruffle doesn’t even run on them (although then, we probably need a different way to determine Windows CE).